### PR TITLE
feat: Lockout policy

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/command/setup_step4.go [hunk 1]
```diff
@@ -24,16 +22,7 @@ func (s *Step4) execute(ctx context.Context, commandSide *Commands) error {
...
-		event, err := c.addDefaultPasswordLockoutPolicy(ctx, iamAgg, NewIAMPasswordLockoutPolicyWriteModel(), &domain.PasswordLockoutPolicy{
-			MaxAttempts:         step.DefaultPasswordLockoutPolicy.MaxAttempts,
-			ShowLockOutFailures: step.DefaultPasswordLockoutPolicy.ShowLockOutFailures,
-		})
-		if err != nil {
-			return nil, err
-		}
-		logging.Log("SETUP-Bfnge").Info("default password lockout policy set up")
-		return []eventstore.EventPusher{event}, nil
+		return nil, nil
```
Comment (adlerhurst) [inline internal/command/setup_step4.go:25]: this looks wrong to me. We should now corrent old setup steps.

File: internal/command/user_human_password.go [hunk 2]
```diff
@@ -225,9 +225,18 @@ func (c *Commands) HumanCheckPassword(ctx context.Context, orgID, userID, passwo
 		_, err = c.eventstore.PushEvents(ctx, user.NewHumanPasswordCheckSucceededEvent(ctx, userAgg, authRequestDomainToAuthRequestInfo(authRequest)))
 		return err
 	}
-	_, err = c.eventstore.PushEvents(ctx, user.NewHumanPasswordCheckFailedEvent(ctx, userAgg, authRequestDomainToAuthRequestInfo(authRequest)))
+	events := make([]eventstore.EventPusher, 0)
+	errMsg := "Errors.User.Password.Invalid"
```
Comment (adlerhurst) [inline internal/command/user_human_password.go:229]: why do you create the error msg here?

File: internal/iam/repository/eventsourcing/model/lockout_policy.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+func (p *LockoutPolicy) Changes(changed *LockoutPolicy) map[string]interface{} {
+	changes := make(map[string]interface{}, 2)
+
+	if p.MaxPasswordAttempts != changed.MaxPasswordAttempts {
+		changes["maxAttempts"] = changed.MaxPasswordAttempts
+	}
+	if p.ShowLockOutFailures != changed.ShowLockOutFailures {
+		changes["showLockOutFailures"] = changed.ShowLockOutFailures
+	}
```
Comment (adlerhurst) [inline internal/iam/repository/eventsourcing/model/lockout_policy.go:36]: it's hard to diff. have we written `LockOut` with upper `O` before?

File: internal/command/user_human_password.go (hunk 2 above)
Comment (fgerschwiler) [reply internal/command/user_human_password.go:229]: bevor I checked on the lockout policy show failure flag and if it wasnt set, I changed the err msg, but I removed this part for the moment

File: internal/iam/repository/eventsourcing/model/lockout_policy.go (hunk 3 above)
Comment (fgerschwiler) [reply internal/iam/repository/eventsourcing/model/lockout_policy.go:36]: Yes I didnt change something on the existing part

File: internal/command/setup_step4.go [hunk 4]
```diff
@@ -24,16 +22,7 @@ func (s *Step4) execute(ctx context.Context, commandSide *Commands) error {
 
 func (c *Commands) SetupStep4(ctx context.Context, step *Step4) error {
```
Comment (adlerhurst) [inline internal/command/setup_step4.go:23]:
```suggestion
//SetupStep4 is removed in favour of step 18.
//Password lockout policy is replaced by lockout policy
func (c *Commands) SetupStep4(ctx context.Context, step *Step4) error {
```

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**